pandas/core/generic.py

```diff
@@ -3430,6 +3432,14 @@ class  max_speed
         1  monkey  mammal        NaN
         3    lion  mammal       80.5
         """
+        if is_copy is not None:
+            warnings.warn(
+                "is_copy is deprecated and will be removed in a future version",
+                FutureWarning,
+                stacklevel=2,
+            )
+        is_copy = True
```

Comment:
Is this assignment even required? Not used later in the function body right?

Reply:
There is a `if is_copy:` a few lines below

Reply:
But, we should only put it to True if it was None (to keep the existing behaviour), so put it in an `else` block ?

Reply:
It is used:
```python
        # Maybe set copy if we didn't actually change the index.
        if is_copy:
            if not result._get_axis(axis).equals(self._get_axis(axis)):
                result._set_is_copy(self)
        return result
```

But is_copy is already deprecated. I think it needs to be removed for 1.0. Since this is going into 1.0 I am thinking now that instead of deprecating this parameter it needs to be removed all together, along with `is_copy` ?